src/util-classification-config.c

```diff
@@ -281,12 +282,10 @@ int SCClassConfAddClasstype(DetectEngineCtx *de_ctx, char *rawstr, uint16_t inde
         SCLogInfo("pcre_copy_substring() failed");
         goto error;
     }
-    if (strlen(ct_priority_str) == 0) {
+    if (strlen(ct_priority_str) == 0 || StringParseInt32(&ct_priority, 10, 0, (const char *)ct_priority_str) < 0) {
```

Comment:
Is this `strlen(ct_priority_str) == 0` check still necessary? How will `StringParseInt32` respond to a 0-length input string?

Reply:
It'll throw an error `Invalid numeric value`. Removing the check.